File: frontend/src/types/course.ts
export interface Course {
  id: number
  user_id: number
  name: string
  description: string
  color: string
  sort_order: number
  created_at: string
  updated_at: string
}

export interface CreateCourseRequest {
  name: string
  description?: string
  color?: string
}

export interface UpdateCourseRequest {
  name?: string
  description?: string
  color?: string
  sort_order?: number
}

File: backend/internal/model/course.go
package model

import "time"

type Course struct {
	ID          uint      `gorm:"primaryKey" json:"id"`
	UserID      uint      `gorm:"index;not null" json:"user_id"`
	Name        string    `gorm:"size:100;not null" json:"name"`
	Description string    `gorm:"size:500" json:"description"`
	Color       string    `gorm:"size:7" json:"color"`
	SortOrder   int       `gorm:"default:0" json:"sort_order"`
	CreatedAt   time.Time `json:"created_at"`
}
